runelite-client/src/main/java/net/runelite/client/plugins/mining/MiningPlugin.java

```diff
@@ -0,0 +1,499 @@
+	}
+
+	@Subscribe
+	public void onGameStateChanged(GameStateChanged event)
+	{
+		if (event.getGameState() == GameState.LOADING ||
+			event.getGameState() == GameState.HOPPING ||
+			event.getGameState() == GameState.LOGGING_IN)
+		{
+			objects.clear();
+			wallObjects.clear();
+		}
+	}
+
+
+	@Subscribe
+	public void onChatMessage(ChatMessage event)
+	{
+		if (event.getType() != ChatMessageType.SPAM)
+		{
+			return;
+		}
+		if (event.getMessage().toLowerCase().contains("you just mined an"))
+		{
+			currentNode = "Gem";
+		}
+		else if (event.getMessage().toLowerCase().contains("you manage to mine"))
+		{
+			currentNode = event.getMessage().replace("You manage to mine some", "").replace(".", "").replaceAll(" ", "");
+		}
+	}
+
+	public String convertObjectToNodeName(int id)
+	{
+		switch (id)
+		{
+			case ObjectID.ROCKS_7463: // Gem
+			case ObjectID.ROCKS_7464: // Gem
+				return "Gem";
+			case ObjectID.ROCKS_7454: // Clay
+			case ObjectID.ROCKS_7487: // Clay
+				return "Clay";
+			case ObjectID.ROCKS_7453: // Copper
+			case ObjectID.ROCKS_7484: // Copper
+				return "Copper";
+			case ObjectID.ROCKS_7485: // Tin
+			case ObjectID.ROCKS_7486: // Tin
+				return "Tin";
+			case ObjectID.ROCKS_7455: // Iron
+			case ObjectID.ROCKS_7488: // Iron
+				return "Iron";
+			case ObjectID.ROCKS_7467: // Granite
+				return "Granite";
+			case ObjectID.ROCKS_8727: // Sandstone
+				return "Sandstone";
+			case ObjectID.ROCKS_7457: // Silver
+			case ObjectID.ROCKS_7490: // Silver
+				return "Silver";
+			case ObjectID.ROCKS_7456: // Coal
+			case ObjectID.ROCKS_7489: // Coal
+				return "Coal";
+			case ObjectID.ROCKS_7458: // Gold
+			case ObjectID.ROCKS_7491: // Gold
+				return "Gold";
+			case ObjectID.ROCKS_7459: // Mithril
+			case ObjectID.ROCKS_7492: // Mithril
+				return "Mithril";
+			case ObjectID.ROCKS_7460: // Adamant
+			case ObjectID.ROCKS_7493: // Adamant
+				return "Adamantite";
+			case ObjectID.ROCKS_7461: // Rune
+			case ObjectID.ROCKS_7494: // Rune
+				return "Runite";
+			case ORE_VEIN_26661: // Motherlode vein
+			case ORE_VEIN_26662: // Motherlode vein
+			case ORE_VEIN_26663: // Motherlode vein
+			case ORE_VEIN_26664: // Motherlode vein
+			case DEPLETED_VEIN_26665: // Depleted motherlode vein
+			case DEPLETED_VEIN_26666: // Depleted motherlode vein
+			case DEPLETED_VEIN_26667: // Depleted motherlode vein
+			case DEPLETED_VEIN_26668: // Depleted motherlode vein
+				return "Pay-dirt";
+			case CRYSTALS: // Amethyst crystals
+			case CRYSTALS_30372: // Amethyst crystals
+			case EMPTY_WALL: // Depleted amethyst crystals
+				return "Amethyst";
+		}
+		return "";
+	}
+
+	@Subscribe
+	public void onGameObjectDespawned(GameObjectDespawned event)
+	{
+		final GameObject object = event.getGameObject();
+		final Player player = client.getLocalPlayer();
+
+		if (object.getWorldLocation().distanceTo(player.getWorldLocation()) > 20)
+		{
+			return;
+		}
+		objects = removeDuplicates(objects);
```

Comment:
What is this for? It doesn't look like it can happen when each time you add a new Obj instance.

Reply:
Was to cleanup the list for debug reasons, forgot to remove before committing.